src/Integration/Binding/JsTsProjectTypeIndicator.cs

```diff
@@ -0,0 +1,116 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Abstractions;
+using System.Linq;
+using EnvDTE;
+using SonarLint.VisualStudio.Infrastructure.VS;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LanguageDetection;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    [Export(typeof(IJsTsProjectTypeIndicator))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class JsTsProjectTypeIndicator : IJsTsProjectTypeIndicator
+    {
+        readonly ISonarLanguageRecognizer sonarLanguageRecognizer;
+        readonly IFolderWorkspaceService folderWorkspaceService;
+        readonly IFileSystem fileSystem;
+
+        public static List<string> filesChecked = new List<string>();
+
+        [ImportingConstructor]
+        public JsTsProjectTypeIndicator(ISonarLanguageRecognizer sonarLanguageRecognizer, IFolderWorkspaceService folderWorkspaceService)
+            : this(sonarLanguageRecognizer, folderWorkspaceService, new FileSystem())
+        {
+
+        }
+
+        internal JsTsProjectTypeIndicator(ISonarLanguageRecognizer sonarLanguageRecognizer, IFolderWorkspaceService folderWorkspaceService, IFileSystem fileSystem)
+        {
+            this.sonarLanguageRecognizer = sonarLanguageRecognizer;
+            this.folderWorkspaceService = folderWorkspaceService;
+            this.fileSystem = fileSystem;
+        }
+        
+        public bool IsJsTs(Project dteProject)
+        {
+            //When opened as folder there can be a dteProject if a file is open
+            //If both are true folder search takes precedense for consistency
+            if (folderWorkspaceService.IsFolderWorkspace())
+            {
+                return HasFolderJSTS();
+            }
+            else
+            {
+                Debug.Assert(dteProject != null, "When it's not folder workspace we expect dteProject not to be null");
+
+                return HasProjectJSTS(dteProject.ProjectItems);
+            }            
+        }
+
+        private bool HasFolderJSTS()
+        {
+            string root = folderWorkspaceService.FindRootDirectory();
+
+            var fileList = fileSystem.Directory.EnumerateFiles(root, "*", SearchOption.AllDirectories).Where(x => !x.Contains("\\node_modules\\"));
```

Comment:
There's no overload that you can only pass the root directory and search options. So I had to pass "*" as a filter. Excluded "node_modules" to prevent having lots of extra files.